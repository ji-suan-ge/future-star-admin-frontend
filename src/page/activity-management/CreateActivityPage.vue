<template>
  <el-container>
    <el-main>
      <el-page-header @back="goBack()" content="新建校友活动"></el-page-header>
      <el-divider/>
      <el-card
        class="activity-card"
        shadow="never">
        <div class="description">- 活动信息 -</div>
        <el-image :src="src"></el-image>
        <el-form
          class="activity-table"
          label-width="150px">
          <div class="line">
            <el-divider></el-divider>
          </div>
          <el-row class="top-distance">
            <el-col :span="8" :offset="offset">
              <i class="el-icon-tickets">&nbsp;活动名称：</i>
              <el-input
                type="text"
                size="medium"
                placeholder="请输入内容"
                v-model="currentActivity.name"
                maxlength="20"
                show-word-limit>
                <template slot="prepend">活动名称:</template>
              </el-input>
            </el-col>
          </el-row>
          <el-row class="top-distance">
            <el-col :span="6" :offset="offset">
              <i class="el-icon-watch">&nbsp;报名开始时间：</i>
              <el-date-picker
                v-model="currentActivity.enroll_start_time"
                type="date"
                placeholder="请选择开始时间">
                <template slot="prepend">开始时间：</template>
              </el-date-picker>
            </el-col>
            <el-col :span="6" :offset="2">
              <i class="el-icon-watch">&nbsp;报名结束时间：</i>
              <el-date-picker
                v-model="currentActivity.enroll_end_time"
                type="date"
                placeholder="请选择结束时间">
                <template slot="prepend">结束时间：</template>
              </el-date-picker>
            </el-col>
          </el-row>
          <el-row class="top-distance">
            <el-col :span="6" :offset="offset">
              <i class="el-icon-watch">&nbsp;活动时间：</i>
              <el-date-picker
                v-model="currentActivity.start_time"
                type="date"
                placeholder="请选择日期时间">
                <template slot="prepend">活动时间：</template>
              </el-date-picker>
            </el-col>
            <el-col :span="6" :offset="2">
              <i class="el-icon-user">&nbsp;面向人群：</i>
              <el-select v-model="clazz" collapse-tags multiple placeholder="请选择">
                <el-option v-for="item in clazzOptions"
                           :key="item.clazz"
                           :label="item.label"
                           :value="item.clazz">
                </el-option>
              </el-select>
            </el-col>
          </el-row>
          <el-row class="top-distance">
            <el-col :span="6" :offset="offset">
              <i class="el-icon-location-information">&nbsp;活动地点：</i>
              <el-input
                type="text"
                size="medium"
                placeholder="请输入内容"
                v-model="currentActivity.address"
                maxlength="20"
                show-word-limit>
                <template slot="prepend">活动地点：</template>
              </el-input>
            </el-col>
            <el-col :span="6" :offset="2">
              <i class="el-icon-crop">&nbsp;限制人数：</i>
              <el-input
                type="text"
                size="medium"
                placeholder="请输入内容"
                v-model="currentActivity.people_number_limit"
                maxlength="3"
                show-word-limit>
                <template slot="prepend">限制人数：</template>
              </el-input>
            </el-col>
          </el-row>
          <el-row class="top-distance">
            <el-col :span="6" :offset="offset">
              <i class="el-icon-money">&nbsp;收费情况：</i>
              <el-input
                type="text"
                size="medium"
                placeholder="请输入内容"
                v-model="currentActivity.price"
                maxlength="4"
                show-word-limit>
                <template slot="prepend">收费情况：</template>
              </el-input>
            </el-col>
            <el-col :span="6" :offset="2">
              <i class="el-icon-s-finance">&nbsp;主办方：</i>
              <el-input
                type="text"
                size="medium"
                placeholder="请输入内容"
                v-model="currentActivity.organizer"
                maxlength="50"
                show-word-limit>
                <template slot="prepend">主办方：</template>
              </el-input>
            </el-col>
          </el-row>
          <el-row class="top-distance">
            <el-col :span="14" :offset="offset">
              <i class="el-icon-fork-spoon">&nbsp;活动日程：</i>
              <el-input
                type="textarea"
                size="medium"
                placeholder="请输入内容"
                v-model="currentActivity.arrangement"
                maxlength="100"
                show-word-limit
                :autosize="{ minRows: 2, maxRows: 4}">
                <template slot="prepend">活动日程：</template>
              </el-input>
            </el-col>
          </el-row>
          <el-row>
            <el-col :span="24" class="button">
              <el-button type="primary" @click="createActivity()">立即创建</el-button>
              <el-button @click="clearText()">清空</el-button>
              <el-button @click="goBack()">取消</el-button>
            </el-col>
          </el-row>
        </el-form>
      </el-card>
    </el-main>
  </el-container>
</template>

<script>
export default {
  name: 'CreateActivityPage',
  data () {
    return {
      page_size: 20,
      currentPage: 1,
      offset: 5,
      clazz: null,
      currentActivity: [],
      clazzOptions: [],
      src: '../assets/backgroundImage1.jpg'
    }
  },
  created () {
    if (this.axios) {
      this.findAllClasses()
    }
  },
  methods: {
    goBack () {
      this.$router.go(-1)
    },
    findAllClasses () {
      let that = this
      that.clazzOptions = []
      that.clazz = null
      let params = {
        page_size: that.page_size,
        page: that.currentClassPage
      }
      that.axios.get('/clazz/clazz', { params }).then((response) => {
        if (response.data.code === '2000') {
          let results = response.data.data.results
          for (let i = 0; i < results.length; i++) {
            that.clazzOptions[i] = {
              clazz: results[i]['id'],
              label: results[i]['name']
            }
          }
          that.$forceUpdate()
        }
      })
    },
    createActivity () {
      let that = this
      let params = {
        name: that.currentActivity.name,
        enroll_start_time: that.currentActivity.enroll_start_time.toLocaleDateString().replace(/\//g, '-'),
        enroll_end_time: that.currentActivity.enroll_end_time.toLocaleDateString().replace(/\//g, '-'),
        organizer: that.currentActivity.organizer,
        start_time: that.currentActivity.start_time.toLocaleDateString().replace(/\//g, '-'),
        address: that.currentActivity.address,
        arrangement: that.currentActivity.arrangement,
        price: that.currentActivity.price,
        people_number_limit: that.currentActivity.people_number_limit
      }
      that.axios.post('/activity/activity', params).then((response) => {
        if (response.data.code === '2000') {
          that.$message({
            type: 'success',
            message: '创建活动成功',
            duration: 2000
          })
          let activityResults = response.data.data
          for (let i = 0; i < that.clazz.length; i++) {
            that.axios.post('/activity/clazz', {
              activity_id: activityResults.id,
              clazz_id: that.clazz[i]
            }).then((response) => {
              if (response.data.code === '2000') {
                that.$message({
                  type: 'success',
                  message: '班级关联活动成功',
                  duration: 2000
                })
              } else {
                that.$message({
                  type: 'error',
                  message: '请求出错',
                  duration: 2000
                })
              }
            }).catch(function (error) {
              console.log(error)
              that.$message({
                type: 'error',
                message: '网络繁忙',
                duration: 2000
              })
            })
          }
        } else {
          that.$message({
            type: 'error',
            message: '请求出错',
            duration: 2000
          })
        }
      }).catch(function (error) {
        console.log(error)
        that.$message({
          type: 'error',
          message: '网络繁忙',
          duration: 2000
        })
      })
    },
    clearText () {
      this.currentActivity.name = ''
      this.currentActivity.enroll_start_time = ''
      this.currentActivity.enroll_end_time = ''
      this.currentActivity.organizer = ''
      this.currentActivity.start_time = ''
      this.currentActivity.address = ''
      this.currentActivity.arrangement = ''
      this.currentActivity.price = ''
      this.currentActivity.people_number_limit = ''
    }
  }
}
</script>

<style scoped>
  .activity-table {
    margin: 40px 20px;
  }

  .description {
    font-size: 21px;
    text-align: center;
    color: black;
  }

  .img {
    text-align: center;
  }

  .el-col > i {
    font-size: 17px;
    color: #272324;
  }

  .el-col > input {
    font-size: 15px;
  }

  .el-col > label {
    margin-bottom: 100px;
    font-size: 14px;
  }

  .top-distance {
    margin-top: 40px;
  }

  .button {
    margin-top: 30px;
    text-align: center;
  }
</style>
