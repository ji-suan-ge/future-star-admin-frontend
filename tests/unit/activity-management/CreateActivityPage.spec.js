import { mount } from '@vue/test-utils'
import CreateActivityPage from '../../../src/page/activity-management/CreateActivityPage'

describe('CreateActivityPage', () => {
  const wrapper = mount(CreateActivityPage)
  test('快照测试', async () => {
    await expect(wrapper.html()).toMatchSnapshot()
  })
  test('是一个 Vue 实例', async () => {
    await expect(wrapper.isVueInstance()).toBeTruthy()
  })
})
