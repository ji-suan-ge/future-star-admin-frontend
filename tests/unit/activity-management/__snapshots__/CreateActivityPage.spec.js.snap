// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateActivityPage 快照测试 1`] = `
<section class="el-container">
  <main class="el-main">
    <div class="el-page-header">
      <div class="el-page-header__left"><i class="el-icon-back"></i>
        <div class="el-page-header__title">返回</div>
      </div>
      <div class="el-page-header__content">新建校友活动</div>
    </div>
    <div class="el-divider el-divider--horizontal">
      <!---->
    </div>
    <div class="el-card activity-card is-always-shadow">
      <!---->
      <div class="el-card__body">
        <form class="el-form activity-table">
          <div class="el-form-item activity-title"><label class="el-form-item__label" style="width: 150px;">活动标题：</label>
            <div class="el-form-item__content" style="margin-left: 150px;">
              <div class="short-text el-input">
                <!----><input type="text" autocomplete="off" class="el-input__inner">
                <!---->
                <!---->
                <!---->
                <!---->
              </div>
              <!---->
            </div>
          </div>
          <div class="el-form-item"><label class="el-form-item__label" style="width: 150px;">活动项目：</label>
            <div class="el-form-item__content" style="margin-left: 150px;">
              <div class="short-text el-input">
                <!----><input type="text" autocomplete="off" class="el-input__inner">
                <!---->
                <!---->
                <!---->
                <!---->
              </div>
              <!---->
            </div>
          </div>
          <div class="el-row">
            <div class="el-col el-col-12">
              <div class="el-form-item"><label class="el-form-item__label" style="width: 150px;">活动时间：</label>
                <div class="el-form-item__content" style="margin-left: 150px;">
                  <div class="el-date-editor el-input el-input--prefix el-input--suffix el-date-editor--datetime">
                    <!----><input type="text" autocomplete="off" name="" placeholder="选择日期时间" class="el-input__inner"><span class="el-input__prefix"><i class="el-input__icon el-icon-time"></i><!----></span><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-input__icon"></i><!----><!----><!----><!----></span>
                    <!----></span>
                    <!---->
                    <!---->
                  </div>
                  <!---->
                </div>
              </div>
            </div>
            <div class="el-col el-col-12">
              <div class="el-form-item"><label class="el-form-item__label" style="width: 150px;">活动地点：</label>
                <div class="el-form-item__content" style="margin-left: 150px;">
                  <div class="short-text el-input">
                    <!----><input type="text" autocomplete="off" class="el-input__inner">
                    <!---->
                    <!---->
                    <!---->
                    <!---->
                  </div>
                  <!---->
                </div>
              </div>
            </div>
          </div>
          <div class="el-row">
            <div class="el-col el-col-8">
              <div class="el-form-item"><label class="el-form-item__label" style="width: 150px;">面向人群：</label>
                <div class="el-form-item__content" style="margin-left: 150px;">
                  <div class="short-text el-input">
                    <!----><input type="text" autocomplete="off" class="el-input__inner">
                    <!---->
                    <!---->
                    <!---->
                    <!---->
                  </div>
                  <!---->
                </div>
              </div>
            </div>
            <div class="el-col el-col-8">
              <div class="el-form-item"><label class="el-form-item__label" style="width: 150px;">限制人数：</label>
                <div class="el-form-item__content" style="margin-left: 150px;">
                  <div class="short-text el-input">
                    <!----><input type="text" autocomplete="off" class="el-input__inner">
                    <!---->
                    <!---->
                    <!---->
                    <!---->
                  </div>
                  <!---->
                </div>
              </div>
            </div>
            <div class="el-col el-col-8">
              <div class="el-form-item"><label class="el-form-item__label" style="width: 150px;">收费情况：</label>
                <div class="el-form-item__content" style="margin-left: 150px;">
                  <div class="short-text el-input">
                    <!----><input type="text" autocomplete="off" class="el-input__inner">
                    <!---->
                    <!---->
                    <!---->
                    <!---->
                  </div>
                  <!---->
                </div>
              </div>
            </div>
          </div>
          <div class="el-form-item"><label class="el-form-item__label" style="width: 150px;">活动内容描述：</label>
            <div class="el-form-item__content" style="margin-left: 150px;">
              <div class="long-text el-textarea"><textarea autocomplete="off" class="el-textarea__inner" style="min-height: -4px;"></textarea>
                <!---->
              </div>
              <!---->
            </div>
          </div>
          <div class="el-form-item"><label class="el-form-item__label" style="width: 150px;">组织单位信息：</label>
            <div class="el-form-item__content" style="margin-left: 150px;">
              <div class="long-text el-textarea"><textarea autocomplete="off" class="el-textarea__inner" style="min-height: -4px;"></textarea>
                <!---->
              </div>
              <!---->
            </div>
          </div>
          <div class="el-form-item"><label class="el-form-item__label" style="width: 150px;">组织单位联系方式：</label>
            <div class="el-form-item__content" style="margin-left: 150px;">
              <div class="long-text el-textarea"><textarea autocomplete="off" class="el-textarea__inner" style="min-height: -4px;"></textarea>
                <!---->
              </div>
              <!---->
            </div>
          </div>
          <div class="el-form-item"><label class="el-form-item__label" style="width: 150px;">报名截止时间：</label>
            <div class="el-form-item__content" style="margin-left: 150px;">
              <div class="el-date-editor el-range-editor el-input__inner el-date-editor--datetimerange"><i class="el-input__icon el-range__icon el-icon-time"></i><input autocomplete="off" placeholder="开始日期" name="" class="el-range-input"><span class="el-range-separator">-</span><input autocomplete="off" placeholder="结束日期" name="" class="el-range-input"><i class="el-input__icon el-range__close-icon"></i></div>
              <!---->
            </div>
          </div>
          <div class="el-form-item">
            <!---->
            <div class="el-form-item__content" style="margin-left: 150px;"><button type="button" class="el-button el-button--primary">
                <!---->
                <!----><span>立即创建</span></button> <button type="button" class="el-button el-button--default">
                <!---->
                <!----><span>取消</span></button>
              <!---->
            </div>
          </div>
        </form>
      </div>
    </div>
  </main>
</section>
`;
